pkg/scoring/gcp_client.go

```
@@ -0,0 +1,215 @@
+	GetOrgIAMPolicy(ctx context.Context, project string) ([]gcpIAMBinding, error)
+	CountAccessibleProjects(ctx context.Context) (int, error)
+}
+
+type gcpClientFactory func(ctx context.Context, key *gcpServiceAccountKey) (gcpAPI, error)
+
+func defaultGCPClientFactory(ctx context.Context, key *gcpServiceAccountKey) (gcpAPI, error) {
+	saJSON, err := key.toJSON()
+	if err != nil {
+		return nil, err
+	}
+	creds, err := google.CredentialsFromJSON(ctx, saJSON,
```

golangci-lint (2.12.2)

[error] SA1019: google.CredentialsFromJSON is deprecated: This function is deprecated because of a potential security risk. It does not validate the credential configuration. The security risk occurs when a credential configuration is accepted from a source that is not under your control and used without validation on your side.

(staticcheck)